oauth2_provider/tests/test_application_views.py

```diff
@@ -76,6 +77,12 @@ def setUp(self):
         self.app_bar_1 = self._create_application('app bar_user 1', self.bar_user)
         self.app_bar_2 = self._create_application('app bar_user 2', self.bar_user)
 
+    def test_get_application(self):
+        """Test the get_application method from the AuthorizationView"""
+        self.assertEqual(self.app_foo_1, AuthorizationView().get_application(self.app_foo_1.client_id))
+        # test twice to test the cache
```

Comment:
Unless you're checking DB query counts or calls to the cache, you aren't truly validating caching behavior.